### PR TITLE
Add editor scrolling sensitivity preference

### PR DESCRIPTION
**Pull request by timtmok**
### Intent
Address #11578 

### Approach
Adds a preference to adjust the scroll sensitivity. Higher results in faster scrolling. The range is 1-200, default 100. The default is set to use the device pixel ratio as the modifier on the ACE default scroll speed.

Preference screenshot:
image

### Automated Tests
None

### QA Notes
Any tests that rely on scroll speed might be affected. The behaviour on Windows should be the same. Linux and MacOS might be different but I believe this change really affects those with high DPI displays. 

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

File: src/cpp/session/resources/schema/user-prefs-schema.json [hunk 1]
```diff
@@ -1618,6 +1618,12 @@
             "title": "Discard pending console input on error",
             "type": "boolean",
             "default": true
+        },
+        "editor_scroll_multiplier": {
+            "description": "A integer value, to set the editor scroll multiplier.",
```
Comment (jgutman) [inline src/cpp/session/resources/schema/user-prefs-schema.json:1623]: Maybe a little parenthetical here to indicate higher means more sensitive or something?

File: src/gwt/src/org/rstudio/studio/client/workbench/prefs/model/UserPrefsAccessor.java [hunk 2]
```diff
@@ -3402,6 +3402,18 @@ public PrefValue<Boolean> discardPendingConsoleInputOnError()
          true);
    }
 
+   /**
+    * A value, 1-100, for the editor scroll speed.
```
Comment (jgutman) [inline src/gwt/src/org/rstudio/studio/client/workbench/prefs/model/UserPrefsAccessor.java:3406]: is this not 1-200?

File: src/gwt/src/org/rstudio/studio/client/workbench/prefs/model/UserPrefsAccessorConstants.java [hunk 3]
```diff
@@ -215,6 +215,14 @@ public interface UserPrefsAccessorConstants extends Constants {
    @DefaultStringValue("The number of spaces to insert when pressing the Tab key.")
    String numSpacesForTabDescription();
 
+   /**
+    * The editor scroll sensitivity.
+    */
+   @DefaultStringValue("Editor scroll sensitivity")
+   String editorScrollMultiplierTitle();
+   @DefaultStringValue("The editor scroll speed")
+   String editorScrollMultiplierDescription();
```
Comment (jgutman) [inline src/gwt/src/org/rstudio/studio/client/workbench/prefs/model/UserPrefsAccessorConstants.java:224]: Again, possibly could use a more descriptive description?

**Review by jgutman — APPROVED** [review]

**Comment by jgutman** [comment]
Also I've been unclear as to whether we need to add electron fixes to news, but probably worth adding to NEWS given the several user reports and the addition of a new user preference

**Comment by jgutman** [comment]
I think we need to add some kind of hover text or question mark in the preferences pane because: Editor scroll sensitivity 100 is not at all clear to the user either what it does or what value it should be set to

File: src/gwt/src/org/rstudio/studio/client/workbench/prefs/model/UserPrefsAccessorConstants.java (hunk 3 above)
Comment (jgutman) [reply src/gwt/src/org/rstudio/studio/client/workbench/prefs/model/UserPrefsAccessorConstants.java:224]: Even just Editor scroll speed or Editor scroll speed sensitivity might be clearer for the title

File: src/cpp/session/resources/schema/user-prefs-schema.json (hunk 1 above)
Comment (timtmok) [reply src/cpp/session/resources/schema/user-prefs-schema.json:1623]: Yes, that would be good!

File: src/gwt/src/org/rstudio/studio/client/workbench/prefs/model/UserPrefsAccessorConstants.java (hunk 3 above)
Comment (timtmok) [reply src/gwt/src/org/rstudio/studio/client/workbench/prefs/model/UserPrefsAccessorConstants.java:224]: I kind of followed how VS Code names their preference. Adding speed into the description is a good idea though. Speed and sensitivity are probably the two common words for scrolling.

**Comment by timtmok** [comment]
> Also I've been unclear as to whether we need to add electron fixes to news, but probably worth adding to NEWS given the several user reports and the addition of a new user preference

It just dawned on me that I missed adding it to the NEWS. This will affect server and desktop in addition to Qt and Electron.

> I think we need to add some kind of hover text or question mark in the preferences pane because: Editor scroll sensitivity 100 is not at all clear to the user either what it does or what value it should be set to

I'll take a look. I thought the description is used for that but I'm not sure where that gets displayed to the user.

**Comment by jmcphers** [comment]
>  I thought the description is used for that but I'm not sure where that gets displayed to the user.

The description is used when the preference is shown in the Command Palette, and also to describe the preference in the auto-generated docs.

https://docs.rstudio.com/ide/server-pro/session_user_settings/session_user_settings.html

**Comment by timtmok** [comment]
Updated the descriptions, NEWS, and added the preference to the command palette.

**Comment by timtmok** [comment]
A bit more of a change but the numeric value widget supports a tooltip on the label now.

**Pull request merged**